tables/output_tables.Rmd

```diff
@@ -13,132 +12,168 @@ library(openxlsx)
+  rename(Tmb_all = tmb) 
 
 # read in tmb coding file, select and rename columns
-tmb_coding <- readr::read_tsv(file.path(data_dir, "pbta-snv-mutation-tmb-coding.tsv")) %>%
-  dplyr::select(Tumor_Sample_Barcode, tmb) %>%
-  dplyr::rename(Kids_First_Biospecimen_ID = Tumor_Sample_Barcode) %>% 
-  dplyr::rename(Tmb_coding = tmb)
+tmb_coding <- read_tsv(tmb_coding_file) %>%
+  select(Tumor_Sample_Barcode, tmb) %>%
+  rename(Kids_First_Biospecimen_ID = Tumor_Sample_Barcode) %>% 
+  rename(Tmb_coding = tmb)
 
 # combine files
-tmb_combined <- full_join(tmb_all, tmb_coding)  %>%
+tmb_combined <- full_join(tmb_all, tmb_coding) %>%
   mutate_all(as.character) %>% 
   replace(is.na(.), "NA")
 ```
 
 ### CNS mutational signatures
 ```{r s2 DNA results part 2}
-cns_mut_list <- readRDS("../analyses/mutational-signatures/results/fitted_exposures_signal-cns_sigfit.rds")
-cns_mean <- cns_mut_list[["mean"]] %>%
-  as.data.frame() %>%
-  tibble::rownames_to_column("Kids_First_Biospecimen_ID")  
+# use the deconstructSigs table, not the sigfit table, for consistency with main text
+# for convenience we can load the file deconstructsigs_exposures_merged.tsv which already has full sample information "merged" in _and_ uses the correct Signal RefSig names
+cns_mut_list <- read_tsv(mutsigs_file)
+
+cns_signatures <- cns_mut_list %>%
+  select(Kids_First_Biospecimen_ID, exposure, signature) %>%
+  # label signatures more clearly as RefSig or other
+  mutate(signature = ifelse(
+    signature == "Other", 
+    "Other signatures",
+    glue::glue("RefSig_{signature}")
+  )) %>%
+  spread(signature, exposure) %>%
+  # order columns so "Other" is at the end
+  select(Kids_First_Biospecimen_ID, contains("RefSig"), everything())
 
 ```
 
 ### Chromothripsis regions per sample
 ```{r s2 DNA results part 3}
-chromothripsis_region_df <- readr::read_tsv("../analyses/chromothripsis/results/chromothripsis_summary_per_sample.txt")  
+chromothripsis_region_df <- readr::read_tsv(chromo_file)  
 
 ```
 
 ### combine S2 table
 ```{r s2 DNA results combined}
-list_s2_table <- list(tmb = tmb_combined,
-                     cns_denovo_mut_sigs = cns_mean,
-                     chromothripsis_events = chromothripsis_region_df
+list_s2_table <- list(tumor_mutation_burden = tmb_combined,
+                      fitted_cns_mutational_signatures = cns_signatures,
+                      chromothripsis_events = chromothripsis_region_df
                      )
-openxlsx::write.xlsx(list_s2_table, 
-                     file.path(results_dir, "TableS2-DNA-results-table.xlsx"),
-                     overwrite=TRUE)
+write.xlsx(list_s2_table, 
+           table_s2_file,
+           overwrite=TRUE)
 ```
 
 ## Table S3: RNA results table
 ### TP53 scores
 ```{r s3 RNA results table p1}
 # get tp53 scores 
-tp53_scores <- readr::read_tsv("../analyses/tp53_nf1_score/results/tp53_altered_status.tsv") %>%
+tp53_scores <- read_tsv(tp53_file) %>%
   mutate_all(as.character) %>% 
```

Comment:
I don't know why this is here?

Reply:
> Something is going on with all the `mutate_all(as.character)` that is happening. 🧐

Yeah, I really didn't know why either. I left them in to avoid changing too much code, but will explore more deeply now.